support/vm/vector.go

```diff
@@ -334,7 +335,7 @@ func (a *adtNodeGetter) Get(ctx context.Context, c cid.Cid) (format.Node, error)
 	if err != nil {
 		return nil, err
 	}
-	return format.Decode(b)
+	return format.Decode(b, ipldcbor.DecodeBlock)
```

Comment:
1. It seems like you were only using dag-cbor here, is that true?
2. Would you find it more helpful to either
     a) just do `ipldcbor.DecodeBlock(b)` instead of using the helper in go-ipld-format for checking if the Block is already a Node?
      b) have a registry object you can create and use (either as a global in this file or passed into the constructor) so that you get `myRegistry.Decode(b)`